python/amici/ode_export.py

```diff
@@ -3344,3 +3350,54 @@ def smart_subs_dict(sym: sp.Expr,
         for eid, expr in subs.items()
         if eid in sym.free_symbols
     })
+
+
+@contextlib.contextmanager
+def _monkeypatched(obj: object, name: str, patch: Any):
+    """
+    Temporarily monkeypatches an object.
+
+    :param obj:
+        object to be patched
+
+    :param name:
+        name of the attribute to be patched
+
+    :param patch:
+        patched value
+
+    """
+    pre_patched_value = getattr(obj, name)
+    setattr(obj, name, patch)
+    try:
+        yield object
+    finally:
+        setattr(obj, name, pre_patched_value)
+
+
+def _custom_pow_eval_derivative(self, s):
+    """
+    Custom Pow derivative that removes a removeable singularity for
+    self.base == 0 and self.base.diff(s) == 0. This function is intended to
+    be monkeypatched into sp.Pow._eval_derivative.
+
+    :param self:
+        sp.Pow class
+
+    :param s:
+        variable with respect to which the derivative will be computed
+
+    """
+    dbase = self.base.diff(s)
+    dexp = self.exp.diff(s)
+    part1 = sp.Pow(self.base, self.exp - 1) * self.exp * dbase
+    part2 = self * dexp * sp.log(self.base)
+    if dbase.is_nonzero or self.base.is_nonzero or dbase.is_nonzero or \
+            part2.is_zero:
+        # first piece never applies or is zero anyways
+        return part1 + part2
+
+    return part1 + sp.Piecewise(
+        (self.base, sp.And(sp.Eq(self.base, 0), sp.Eq(dbase, 0))),
```

Comment:
```suggestion
        (sp.Float(0), sp.And(sp.Eq(self.base, 0), sp.Eq(dbase, 0))),
```
Perhaps more readable?

Reply:
True, but we do not know whether `self.base` is `sp.Integer(0)` or `sp.Float(0)` or some other variable that is equal zero and for which dbase also happens to be zero, so I think returning `self.base` is the implementation that makes the least possible assumptions.

Reply:
The only downside of using `base` I see is a bit more code bloat (since it may be a *long* expression), but performance-wise the compiler should optimize it away.